cmd/qat_plugin/README.md

```diff
@@ -1,39 +1,211 @@
+|:-------- |:-------- |:--------------- |:----------- |
+| -dpdk-driver | `$DPDK_DRIVER` | vfio-pci | A more robust and secure choice than the `igb_uio` alternative |
+| -kernel-vf-drivers | `$KERNEL_VF_DRIVERS` | dh895xccvf,c6xxvf,c3xxxvf,d15xxvf | Modify to suit your hardware setup |
+| -max-num-devices | `$MAX_NUM_DEVICES` | 32 | Modify to suit your hardware setup if necessary |
+| -debug | `$DEBUG` | false | Debug is disabled by default |
+
+For more details on the `-dpdk-driver` choice, see 
+[DPDK Linux Driver Guide](http://dpdk.org/doc/guides/linux_gsg/linux_drivers.html).
+
+For more details on the available options to the `-kernel-vf-drivers` option, see the list of
+vf drivers available in the [Linux Kernel](https://github.com/torvalds/linux/tree/master/drivers/crypto/qat).
+
+> **Note**: `kerneldrv` mode is excluded from builds by default. See the
+> [Build the plugin image](#build-the-plugin-image) section for more details.
+
+If the `-mode` parameter is set to `kerneldrv`, no other parameter documented above are valid,
+except `-debug` which is global for both modes.
+`kerneldrv` mode implements resource allocation based on system configured [logical instances][7].
+
+The `kerneldrv` mode does not guarantee full device isolation between containers
+and therefore it's not recommended. This mode will be deprecated and removed once `libqat`
+implements non-UIO based device access.
+
+# Installation
+
+The below sections cover how to obtain, build and install this component.
+
+The component can be installed either using a DaemonSet or running 'by hand' on each node.
+
+## Pre-built image
+
+[Pre-built images](https://hub.docker.com/r/intel/intel-qat-plugin)
+of this component are available on the Docker hub. These images are automatically built and uploaded
+from the latest master branch of this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers.
+
+The deployment YAML files supplied with this component in this repository use the images with the `devel`
+tag by default. If you do not build your own local images, your Kubernetes cluster may pull down
+the devel images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the
+[YAML deployment files](../../deployments/qat_plugin/)
+appropriately.
+
+## Prerequisites
+
+The component has the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about).
+
+You will also need [appropriate hardware installed](#checking-for-hardware).
+
+The QAT plugin requires Linux Kernel VF FPGA drivers to be available. These drivers
+are available via two methods. One of them must be installed and enabled:
+
+- [Linux Kernel upstream drivers](https://github.com/torvalds/linux/tree/master/drivers/crypto/qat)
+- [Intel QuickAssist Technology software for Linux][9]
+
+The demonstrations have their own requirements, listed in their own specific sections.
+
```

Comment:
a note about `-dpdk-driver`:  `vfio-pci` is part of the linux kernel while `igb_uio` comes with DPDK sources and must be built separately against the running kernel. That is also a DPDK specific thing. With Kata for instance we use the VFIO PCI device assignment and `vfio-pci`.

Reply:
not sure how to place that in the docs - will leave for you to append in a future PR?

Reply:
ok!